torch/nested/_internal/nested_tensor.py

```diff
@@ -1,23 +1,49 @@
 # mypy: allow-untyped-defs
+from contextlib import contextmanager
 from typing import Tuple
 
 import torch
 from torch._C import DispatchKey, DispatchKeySet
 from torch._prims_common import is_expandable_to
-from torch.fx.experimental.symbolic_shapes import has_free_symbols
 from torch.utils.weak import WeakTensorKeyDictionary
 from typing import *  # noqa: F403
 
 _tensor_id_counter = 0
 _tensor_symint_registry = WeakTensorKeyDictionary()
 
 
+@contextmanager
+def freeze_nested_int_id_counter():
+    global _tensor_id_counter
+    orig_counter = _tensor_id_counter
+    try:
+        yield
+    finally:
+        _tensor_id_counter = orig_counter
```

Comment:
I made some updates to the other PR btw, and one of the changes was adding a new separate tensor_id_counter specific for tracing  to avoid polluting any eager state while tracing (more info in the newly beefed up PR descr)

For now I'm storing it on FakeTensorMode and resetting based on the value of the eager id counter upon enter/exit, but there are probably more subtleties I'm missing.

Reply:
ooh I like this much better! lemme go check it out